doc/en/monkeypatch.rst

```diff
@@ -154,6 +330,112 @@ This behavior can be moved into ``fixture`` structures and shared across tests:
             _ = get_os_user_lower()
 
 
+Monkeypatching dictionaries
+---------------------------
+
+:py:meth:`monkeypatch.setitem` can be used to safely set the values of dictionaries
+to specific values during tests. Take this simplified connection string example:
+
+.. code-block:: python
+
+    # Contents of app.py to generate a simple connection string.
+    DEFAULT_CONFIG = {"user": "user1", "database": "db1"}
+
+
+    def create_connection_string(config=None):
+        """Creates a connection string from input or defaults."""
+        config = config or DEFAULT_CONFIG
+        return f"User Id={config['user']}; Location={config['database']};"
+
+For testing purposes we can patch the ``DEFAULT_CONFIG`` dictionary to specific values.
+
+.. code-block:: python
+
+    # Contents of test_app.py.
+    # Our app.py with the connection string function (prior code block).
+    import app
+
+
+    def test_connection(monkeypatch):
+
+        # Patch the values of DEFAULT_CONFIG to specific
+        # testing values only for this test.
+        monkeypatch.setitem(app.DEFAULT_CONFIG, "user", "test_user")
+        monkeypatch.setitem(app.DEFAULT_CONFIG, "database", "test_db")
+
+        # The expected result based on the mocks.
+        expected = "User Id=test_user; Location=test_db;"
+
+        # The test uses the monkeypatched dictionary settings.
+        result = app.create_connection_string()
+        assert result == expected
+
+You can use the :py:meth:`monkeypatch.delitem` to remove values.
+
+.. code-block:: python
+
+    # Contents of test_app.py.
+    import pytest
+
+    # Our app.py with the connection string function.
+    import app
+
+
+    def test_missing_user(monkeypatch):
+
+        # Patch the DEFAULT_CONFIG t be missing the 'user' key.
+        monkeypatch.delitem(app.DEFAULT_CONFIG, "user", raising=False)
+
+        # Key error expected because a config is not passed, and the
+        # default is now missing the 'user' entry.
+        with pytest.raises(KeyError):
+            _ = app.create_connection_string()
+
+
+The modularity of the ``fixture`` syntax gives you the flexibility to define
```

Comment:
```suggestion
The modularity of fixtures gives you the flexibility to define
```

Reply:
Applied in https://github.com/pytest-dev/pytest/commit/7fa0e0405d4c051d176ba40ae4205dc3393e27a7 - also realized the second test does not require the `mock_test_database` fixture so I removed it from the signature - https://github.com/pytest-dev/pytest/commit/7fa0e0405d4c051d176ba40ae4205dc3393e27a7#diff-425daef754438812ec1a0e4d3aeaeff8L434